src/Compilers/CSharp/Portable/Binder/Binder_Initializers.cs

```diff
@@ -161,10 +161,7 @@ internal Binder GetFieldInitializerBinder(FieldSymbol fieldSymbol, bool suppress
 
             Binder binder = this;
 
-            if (!fieldSymbol.IsStatic && fieldSymbol.ContainingType.GetMembersUnordered().OfType<SynthesizedPrimaryConstructor>().SingleOrDefault() is SynthesizedPrimaryConstructor primaryCtor)
-            {
-                binder = new InMethodBinder(primaryCtor, binder);
-            }
+            binder = new WithPrimaryConstructorParametersBinder(fieldSymbol.ContainingType, binder);
```

Comment:
>new WithPrimaryConstructorParametersBinder(fieldSymbol.ContainingType, binder); [](http://example.com/codeflow?start=21&length=79)

Should this binder be added only when the containing type has a primary constructor?

Reply:
> Should this binder be added only when the containing type has a primary constructor?

This binder is with lazy behavior, it handles the situation when there is no primary constructor. This helps avoid a cycle in the process of constructing members. We need to bind some types while members are still being built

Reply:
Perhaps in this particular place we could duplicate the conditions, but I do not think it is worth it.